### PR TITLE
chore: add configuration for datapipeline

### PR DESCRIPTION
**Pull request by Shahroz16**
closes: https://linear.app/customerio/issue/MBL-219/update-the-config-for-initialization

- Updates to `DataPipelineModule` and `CustomerIOBuilder` to add the missing configs
- Added tests
- Moved `CustomerIOBuilderTest` to unit test to speed up

### CODE REVIEW

File: core/src/main/kotlin/io/customer/sdk/data/model/Region.kt
Comment (Shahroz16) [inline core/src/main/kotlin/io/customer/sdk/data/model/Region.kt:1]: its in core now

File: datapipelines/build.gradle [hunk 1]
```diff
@@ -37,4 +37,8 @@ dependencies {
     api project(":core")
     implementation(Dependencies.segment)
     implementation(Dependencies.kotlinxSerializationJson)
+
+    api Dependencies.mockito
+    api Dependencies.mockitoKotlin
+    api Dependencies.robolectric
```
Comment (Shahroz16) [inline datapipelines/build.gradle:43]: need roboelectric for application instance in unit test

File: datapipelines/src/main/kotlin/io/customer/datapipelines/DataPipelinesModule.kt [hunk 2]
```diff
@@ -18,13 +23,30 @@ internal constructor(
 ) : CustomerIOModule<DataPipelinesModuleConfig>, CustomerIOInstance {
     override val moduleName: String = MODULE_NAME
 
-    private val analytics: Analytics
+    private val errorLogger = object : ErrorHandler {
```
Comment (Shahroz16) [inline datapipelines/src/main/kotlin/io/customer/datapipelines/DataPipelinesModule.kt:26]: to put logs under our tag

File: datapipelines/build.gradle (hunk 1 above)
Comment (mrehan27) [inline datapipelines/build.gradle:43]: Shouldn't these dependencies be `testImplementation`?

```suggestion
    testImplementation(Dependencies.mockito)
    testImplementation(Dependencies.mockitoKotlin)
    testImplementation(Dependencies.robolectric)
```

File: datapipelines/src/main/kotlin/io/customer/datapipelines/DataPipelinesModule.kt (hunk 2 above)
Comment (mrehan27) [reply datapipelines/src/main/kotlin/io/customer/datapipelines/DataPipelinesModule.kt:26]: Can you please add this as comment?

File: datapipelines/src/main/kotlin/io/customer/datapipelines/DataPipelinesModule.kt [hunk 3]
```diff
@@ -18,13 +23,30 @@ internal constructor(
...
+    }
+
+    private val analytics: Analytics = Analytics(
+        writeKey = moduleConfig.cdpApiKey,
+        context = androidSDKComponent.applicationContext
+    ) {
+        flushAt = moduleConfig.flushAt
+        flushInterval = moduleConfig.flushInterval
+        flushPolicies = moduleConfig.flushPolicies
+        autoAddSegmentDestination = false
```
Comment (mrehan27) [inline datapipelines/src/main/kotlin/io/customer/datapipelines/DataPipelinesModule.kt:39]: Would be great if can leave a comment here why keeping it `false` is important.

File: datapipelines/src/test/java/io/customer/datapipelines/sdk/CustomerIOBuilderTest.kt [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+import org.mockito.kotlin.never
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.robolectric.Robolectric
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class CustomerIOBuilderTest : BaseUnitTest() {
+
+    private val application: Application = Robolectric.buildActivity(Activity::class.java).setup().get().application
```
Comment (mrehan27) [inline datapipelines/src/test/java/io/customer/datapipelines/sdk/CustomerIOBuilderTest.kt:28]: I think this should be moved to some base class as we'll keep needing this for several tests. I'm okay to not changes it in this PR, we can cover this in Setup Testing Environment ticket. But adding note so we don't miss this.

File: datapipelines/src/test/java/io/customer/datapipelines/sdk/CustomerIOBuilderTest.kt [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+import org.junit.runner.RunWith
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.never
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.robolectric.Robolectric
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class CustomerIOBuilderTest : BaseUnitTest() {
```
Comment (mrehan27) [inline datapipelines/src/test/java/io/customer/datapipelines/sdk/CustomerIOBuilderTest.kt:26]: I think there are a few tests missing:

- Validating default values
- Validating setting regions (both `US` and `EU`) works correctly and as expected

**Review by mrehan27 — COMMENTED** [review]
Looks mostly good, but I think we are missing some important tests. Once added, I feel we are good to merge this.

File: datapipelines/build.gradle (hunk 1 above)
Comment (Shahroz16) [reply datapipelines/build.gradle:43]: 🤦

Comment (Shahroz16) [reply datapipelines/build.gradle:43]: 🤦

File: datapipelines/src/test/java/io/customer/datapipelines/sdk/CustomerIOBuilderTest.kt (hunk 4 above)
Comment (Shahroz16) [reply datapipelines/src/test/java/io/customer/datapipelines/sdk/CustomerIOBuilderTest.kt:28]: This is robolectric specific, not sure if its wise to make our base test robolectric, either we create another robolectric base class that extends base class or let each class add this if needed.

Comment (mrehan27) [reply datapipelines/src/test/java/io/customer/datapipelines/sdk/CustomerIOBuilderTest.kt:26]: Yes, I meant to have another base class to help similar cases, we can look into this later with test changes though, no action required for this PR.

**Review by mrehan27 — APPROVED** [review]
Looks good now 👍🏻

**Pull request merged**